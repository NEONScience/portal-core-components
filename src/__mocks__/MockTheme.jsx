import React from 'react';

import CssBaseline from '@mui/material/CssBaseline';
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles';

import Theme from '../lib_components/components/Theme/Theme';

const MockTheme = ({ children }) => {
  return (
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={Theme}>
        <CssBaseline />
        {children}
      </ThemeProvider>
    </StyledEngineProvider>
  );
}

export default MockTheme;
