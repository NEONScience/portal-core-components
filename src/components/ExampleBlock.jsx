import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';

import Container from '@material-ui/core/Container';

import Theme from '../lib_components/components/Theme/Theme';

const useStyles = makeStyles(theme => ({
  root: {
    margin: theme.spacing(2, 0),
    padding: theme.spacing(3),
    backgroundColor: theme.palette.grey[50],
    borderRadius: theme.spacing(1),
    display: 'flex',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
}));

export default function ExampleBlock(props) {
  const classes = useStyles(Theme);
  const { children } = props;
  return (
    <Container className={classes.root}>
      {children}
    </Container>
  );
}

ExampleBlock.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.oneOfType([
      PropTypes.node,
      PropTypes.string,
    ])),
    PropTypes.node,
    PropTypes.string,
  ]).isRequired,
};
