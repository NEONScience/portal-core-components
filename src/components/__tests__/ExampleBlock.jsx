import React from 'react';
import renderer from 'react-test-renderer';

import MockTheme from '../../__mocks__/MockTheme';
import ExampleBlock from '../ExampleBlock';

describe('ExampleBlock', () => {
  test('renders with a single child', () => {
    const tree = renderer
      .create(
        <MockTheme>
          <ExampleBlock>
            <div>1</div>
          </ExampleBlock>
        </MockTheme>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders with many children (row)', () => {
    const tree = renderer
      .create(
        <MockTheme>
          <ExampleBlock>
            <div>1</div>
            <div>2</div>
            <div>3</div>
          </ExampleBlock>
        </MockTheme>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders with many children (column)', () => {
    const tree = renderer
      .create(
        <MockTheme>
          <ExampleBlock column>
            <div>4</div>
            <div>5</div>
            <div>6</div>
          </ExampleBlock>
        </MockTheme>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
