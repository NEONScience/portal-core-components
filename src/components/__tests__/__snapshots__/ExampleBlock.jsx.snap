// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleBlock renders with a single child 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg makeStyles-root-1 css-1oqqzyl-MuiContainer-root"
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <div>
    1
  </div>
</div>
`;

exports[`ExampleBlock renders with many children (column) 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg makeStyles-root-3 css-1oqqzyl-MuiContainer-root"
  style={
    {
      "flexDirection": "column",
    }
  }
>
  <div>
    4
  </div>
  <div>
    5
  </div>
  <div>
    6
  </div>
</div>
`;

exports[`ExampleBlock renders with many children (row) 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg makeStyles-root-2 css-1oqqzyl-MuiContainer-root"
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <div>
    1
  </div>
  <div>
    2
  </div>
  <div>
    3
  </div>
</div>
`;
